api/proto/teleport/legacy/types/types.proto

```diff
@@ -6395,6 +6395,9 @@ message AWSMatcher {
   // Integration is the integration name used to generate credentials to interact with AWS APIs.
   // Environment credentials will not be used when this value is set.
   string Integration = 7 [(gogoproto.jsontag) = "integration,omitempty"];
+  // KubeAppDiscovery controls whether Kubernetes App Discovery will be enabled for agents running on
+  // discovered clusters, currently only affects AWS EKS discovery in integration mode.
+  bool KubeAppDiscovery = 8 [(gogoproto.jsontag) = "kube_app_discovery,omitempty"];
```

Comment:
Do we plan on adding support for more resources? Or other knobs for Kube Agents?
If yes, I would probably create a KubeDiscoveryConfig type and then populate with those fields (only app discovery for now)

Reply:
We currently don't have any plans for anything like that.

Reply:
Since we don't have any plans in that area, I thought it might just be premature optimization/overcomplication, so I chose the smallest scope - just to put flag into aws matcher. As an option we already have type `KubernetesMatcher`, used for standalone Kubernetes App Discovery configuration:
![image](https://github.com/gravitational/teleport/assets/964862/e234c70d-8922-49a5-b385-25a971ab8ce2)
It is now a field on the same level as aws matchers in the discovery config, but we could also add it as subfield into aws matcher and fill in default config (types:["app"], namespaces: "*", labels: "*:*"). But nothing fundamentally stop us from starting small now with just  `KubeAppDiscovery` flag (being shortcut to the default confiig) and then adding that field with config later if we need it.